bigtable-hbase-mirroring-client-1.x-parent/bigtable-hbase-mirroring-client-1.x/src/main/java/com/google/cloud/bigtable/mirroring/hbase1_x/MirroringTable.java

```diff
@@ -0,0 +1,1052 @@
+      Log.trace("[%s] append(append=%s)", this.getName(), append);
+
+      Result result =
+          this.mirroringTracer.spanFactory.wrapPrimaryOperation(
+              new CallableThrowingIOException<Result>() {
+                @Override
+                public Result call() throws IOException {
+                  return MirroringTable.this.primaryTable.append(append);
+                }
+              },
+              HBaseOperation.APPEND);
+
+      Put put = makePutFromResult(result);
+
+      scheduleSequentialWriteOperation(
+          new WriteOperationInfo(put), this.secondaryAsyncWrapper.put(put));
+      return result;
+    }
+  }
+
+  @Override
+  public Result increment(final Increment increment) throws IOException {
+    // TODO: bug - we should force increment to return results
+    // increment.setReturnResults(true);
+    try (Scope scope = this.mirroringTracer.spanFactory.operationScope(HBaseOperation.INCREMENT)) {
+      Log.trace("[%s] increment(increment=%s)", this.getName(), increment);
+
+      Result result =
+          this.mirroringTracer.spanFactory.wrapPrimaryOperation(
+              new CallableThrowingIOException<Result>() {
+                @Override
+                public Result call() throws IOException {
+                  return MirroringTable.this.primaryTable.increment(increment);
+                }
+              },
+              HBaseOperation.INCREMENT);
+
+      Put put = makePutFromResult(result);
+
+      scheduleSequentialWriteOperation(
+          new WriteOperationInfo(put), this.secondaryAsyncWrapper.put(put));
+      return result;
+    }
+  }
+
+  @Override
+  public long incrementColumnValue(byte[] row, byte[] family, byte[] qualifier, long amount)
+      throws IOException {
+    Log.trace(
+        "[%s] incrementColumnValue(row=%s, family=%s, qualifier=%s, amount=%s)",
+        this.getName(), row, family, qualifier, amount);
+    Result result = increment((new Increment(row)).addColumn(family, qualifier, amount));
+    Cell cell = result.getColumnLatestCell(family, qualifier);
+    assert cell != null;
+    return Bytes.toLong(CellUtil.cloneValue(cell));
+  }
+
+  @Override
+  public long incrementColumnValue(
+      byte[] row, byte[] family, byte[] qualifier, long amount, Durability durability)
+      throws IOException {
+    Log.trace(
+        "[%s] incrementColumnValue(row=%s, family=%s, qualifier=%s, amount=%s, durability=%s)",
+        this.getName(), row, family, qualifier, amount, durability);
+    Result result =
+        increment(
+            (new Increment(row)).addColumn(family, qualifier, amount).setDurability(durability));
+    Cell cell = result.getColumnLatestCell(family, qualifier);
+    assert cell != null;
+    return Bytes.toLong(CellUtil.cloneValue(cell));
+  }
+
+  @Override
+  public CoprocessorRpcChannel coprocessorService(byte[] bytes) {
+    throw new UnsupportedOperationException();
+  }
```

Comment:
might be a good idea to group all of the unsupported operation below

Reply:
Done.